bluefog/common/common.h

```diff
@@ -209,10 +209,10 @@ class TensorShape {
 
 class Tensor {
  public:
-  virtual const DataType dtype() const = 0;
+  virtual DataType dtype() const = 0;
```

Comment:
Because here DataType is a return value. DataType is not a pointer or reference, no need to return a const value. This is also a warning provided by the static C++ checker on my side.